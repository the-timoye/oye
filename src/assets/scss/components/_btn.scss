.btn {
    display: inline-block;
    padding: .4rem 2rem;
    border: 1px solid $beige;
    border-radius: 20px;
    text-align: center;
    margin: 0 .5rem;
    transition: all .3s ease-in-out;

    &-primary {

        color: $light_green;
        background: $green;
        &:hover {        
            background: $beige;
            color: $green;
        }
    }
    &-secondary {
        background: $light_green;
        color: $green;
        &:hover {        
            background: $beige;
            
        }
    }
}