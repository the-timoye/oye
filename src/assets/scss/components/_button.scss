.button {
    padding: 0.5rem 2.5rem;
    font-weight: 400;
    border-radius: 5px;

    &-secondary {
        border: 1px solid $primary;
        background-color: transparent;
        color: $primary;
    }
}
