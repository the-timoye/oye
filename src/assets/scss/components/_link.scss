a.link {
    display: inline-block;
    margin: 1rem auto;
    padding: 0.5rem 0;

    hr.link__hr {
        margin-top: 0.5rem;
        border: 0;
        height: 1px;

        &.__light {
            background-image: linear-gradient(to right, rgba(0, 0, 0, 0), $text, rgba(0, 0, 0, 0));
        }
        &.__dark {
            background-image: linear-gradient(to right, rgba(0, 0, 0, 0), $primary, rgba(0, 0, 0, 0));
        }
    }

    &.__light {
        color: $text;
    }
    &.__dark {
        color: $primary;
    }
}
