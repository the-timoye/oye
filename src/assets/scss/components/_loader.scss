.lds-ripple {
    display: inline-block;
    position: absolute;
    top: 50%;
    left: 50%;
    width: 80px;
    height: 80px;

    div {
        position: absolute;
        border: 4px solid $text;
        opacity: 1;
        border-radius: 50%;
        animation: lds-ripple 1s cubic-bezier(0, 0.2, 0.8, 1) infinite;

        &:nth-child(2) {
            animation-delay: -0.5s;
        }
    }
}
