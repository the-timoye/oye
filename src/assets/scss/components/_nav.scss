nav {
    display: flex;
    justify-content: space-between;
    padding: 2rem 1.5rem;

    img {
        height: 3rem;
    }

    div {
        & :not(:last-child) {
            margin-right: 1rem;
        }
        a {
            cursor: pointer;
        }
    }
}
