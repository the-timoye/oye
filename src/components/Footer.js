import React from 'react';
import chatIcon from '../assets/images/chat.png';

export const Footer = () => {
    return (

        <footer className="footer">
            <span>Copyright &#169; 2020</span>
            <span>Designed and Developed by <a href="/"> thetimoye </a> &#9825;</span>
            <img src={chatIcon} alt='chat' />
        </footer>

    )
}