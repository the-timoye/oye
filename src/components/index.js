export * from './Footer';
export * from "./Header";
export * from "./Hobbies";
export * from "./Nav";
export * from "./Projects";
export * from "./SectionTitle"
export * from "./Skills";