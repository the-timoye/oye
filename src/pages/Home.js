import user from '../assets/images/svgs/user.svg'

// import database from '../assets/images/svgs/database.svg'

import ig from '../assets/images/svgs/ig.svg'
import book from '../assets/images/svgs/book.svg'
import github from '../assets/images/svgs/github.svg'
import linkedin from '../assets/images/svgs/linkedin.svg'
import medium from '../assets/images/svgs/medium.svg'

const Home = () => {
    return (
        <div className="home">
            <div className="side_deco" />
            <div className="home__profile-card">
                <div className="home__profile-card__avatar">
                    <img src={user} alt='avatar' />
                </div>
                <h1 className="home__profile-card__name">Timi Oluyinka</h1>
                {/* <div>
                <div className='home__profile-card__dot' /> 
                </div> */}
                <span className="home__profile-card__profession"> Data Engineer</span>
                <div className='home__profile-card__footer'>
                <img src={linkedin} alt='avatar' />
                <img src={medium} alt='avatar' />
                <img src={github} alt='avatar' />
                <img src={ig} alt='avatar' />
                <img src={book} alt='avatar' />
                </div>
            </div>
            <div className="home__profile intro">
                <div className="content">
                <span className="home__profile intro__main-text">Hello
                    <small className="home__profile intro__sub-text">I am not your friend</small>
                </span>

                    <div className="home__profile intro__links">
                        <a className="btn btn-primary" href='#'>Button A</a>
                        <a className="btn btn-secondary" href='#'>Button B</a>
                    </div>

                    <p className="home__profile intro__text">
                    I'm a paragraph. Click here to add your own text and edit me. It’s easy. Just click “Edit Text” or double click me to add your own content and make changes to the font.
                    <br /> I’m a great place for you to tell a story and let your users know a little more about you.
                    </p>
                </div>

            </div>
        </div>
    )
}

export default Home;