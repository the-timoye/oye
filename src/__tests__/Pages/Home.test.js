import React from 'react';
import renderer from 'react-test-renderer';
import { Nav } from '../../components';
import Home from "../../pages/Home";

describe('Home page tests', () => {

    const HomeInstance = renderer.create(<Home />);

    it('renders correctly', () => {
        expect(HomeInstance).toMatchSnapshot();
    });

    it('has a navigation bar', () => {
        const NavInstance = HomeInstance.root.findByType(Nav);
        expect(NavInstance).toBeTruthy();
    });

    it('has an about section', () => {
        const AboutInstance = HomeInstance.root.findByProps({ className: 'about' });
        expect(AboutInstance).toBeTruthy();
    });

    it('has a projects section', () => {
        const ProjectsInstance = HomeInstance.root.findByProps({ className: 'project' });
        expect(ProjectsInstance).toBeTruthy();
    });

    it('has a hobbies section', () => {
        const HobbiesInstance = HomeInstance.root.findByProps({ className: 'hobbies' });
        expect(HobbiesInstance).toBeTruthy();
    });

    it('has a footer', () => {
        const FooterInstance = HomeInstance.root.findByProps({ className: 'footer' });
        expect(FooterInstance).toBeTruthy();
    });

});
