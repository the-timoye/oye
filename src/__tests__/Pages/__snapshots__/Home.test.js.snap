// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page tests renders correctly 1`] = `
Array [
  <nav>
    <img
      alt="logo"
      className="logo"
      src="logo.png"
    />
    <div>
      <a
        href="Oluyinka Oluwatimilehin.pdf"
        onClick={[Function]}
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="file"
          src="file.png"
        />
      </a>
      <a
        href="mailto:thetimoye@gmail.com"
      >
        <img
          alt="mail"
          src="mail.png"
        />
      </a>
    </div>
  </nav>,
  <header
    className="home"
  >
    <div
      className="intro"
    >
      <h2
        className="mb-5"
      >
         
        
         
      </h2>
      <p>
         
        I’m a Software Developer with deep love for code, and everything there is to know about Software Development and Data Engineering
         
      </p>
      <a
        href="https://github.com/the-timoye"
        rel="noreferrer"
        target="_blank"
      >
         
        <img
          alt="Git Hub"
          className="socials"
          src="github.png"
        />
         
      </a>
      <a
        href="https://instagram.com/thetimoye"
        rel="noreferrer"
        target="_blank"
      >
         
        <img
          alt="Instagram"
          className="socials"
          src="instagram.png"
        />
         
      </a>
      <a
        href="https://repl.it/@thetimoye"
        rel="noreferrer"
        target="_blank"
      >
         
        <img
          alt="Repl It"
          className="socials"
          src="replit.png"
        />
         
      </a>
    </div>
    <img
      alt="home-header"
      className="banner"
      src="header.png"
    />
  </header>,
  <div
    className="section-title"
  >
    <hr
      className="section-title__hr"
    />
    <h3>
       
      About Me
       
    </h3>
    <hr
      className="section-title__hr"
    />
  </div>,
  <div
    className="about"
  >
    <p
      className="about"
    >
      I began software development with the normal basics HTML and CSS shortly after graduation, before "Nigeria called". Soon after service, I worked 14months in a tech company as a System's Administrator, and alongside took part in Andela's #ALCWithUdacity2017 scholarship program, and brushed up my Javascript skills. I'm currently a full stack dev in Summitech Computing Ltd. See below my prowess so far:
    </p>
    <div
      className="d-flex flex-wrap justify-content-around"
    />
  </div>,
  <div
    className="section-title"
  >
    <hr
      className="section-title__hr"
    />
    <h3>
       
      Projects
       
    </h3>
    <hr
      className="section-title__hr"
    />
  </div>,
  <div
    className="project"
  />,
  <div
    className="section-title"
  >
    <hr
      className="section-title__hr"
    />
    <h3>
       
      Hobbies
       
    </h3>
    <hr
      className="section-title__hr"
    />
  </div>,
  <div
    className="hobbies"
  >
    <p
      className="my-5 text-center"
    >
       
      Asides Software development, enjoy reading novels (Christian, History and Science fictions), photography (for my love for nature) & browsing through art works
       
    </p>
    <div
      className="hobbies-content"
    >
      <div
        className="hobbies-content__books"
      >
        <p
          className="mb-5"
        >
           Some of my favorite books are: 
        </p>
      </div>
      <div
        className="hobbies-content__photos"
      >
        <p
          className="mb-5"
        >
           Some Images I shot (from my love for art) 
        </p>
        <br />
        <a
          className="link __light float-right"
          href="https://www.instagram.com/thetimoye"
          rel="noreferrer"
          target="_blank"
        >
          see more →
          <hr
            className="link__hr __light"
          />
        </a>
      </div>
    </div>
  </div>,
  <footer
    className="footer"
  >
    <span>
      Copyright © 2020
    </span>
    <span>
      Designed and Developed by 
      <a
        className="link __light"
        href="/"
      >
         thetimoye 
        <hr
          className="link__hr __light"
        />
      </a>
       ♡
    </span>
    <img
      alt="chat"
      src="chat.png"
    />
  </footer>,
]
`;
